webapp/src/components/form/FormMultiFile.js

```diff
@@ -0,0 +1,130 @@
+        }
+    }
+
+    componentWillMount() {
+        if (this.props.value) {
+            this.setState({
+                fileList: this.props.value,
+            })
+        }
+    }
+
+
+    // add file
+    addFile = () => {
+        // variables
+        let condition = true;
+        let addError = false;
+        let handleList = this.state.fileList;
+        // check against empty fields
+        handleList.map(val => {
+            if (!val.file) {
+                condition = false
+                addError = true
+            }
+        })
+        // add item if there is no empty fields
+        if (condition) {
+            handleList.push({ name: null, file: null, delete: null, filePath: null })
+        }
+
+        this.setState({
+            addError: addError,
+            fileList: handleList
+        }, () => {
+            condition = true;
+            addError = false
+        })
+    }
+
+
+    //handle upload
+    handleUpload = (file, name, del, filePath) => {
+        // function variables
+        let handleDuplicates = false;
+        let handleList = this.state.fileList;
+
+        if (del) {
+            if (handleList.length > 1) {
+                // del file
+                let filteredList = handleList.filter((val) => {
+                    return file != val.file
+                })
+                handleList = filteredList
+            }
+            else {
+                handleList = []
+            }
+
+        }
+
+        else {
+            // Add new value
+            handleList.map(val => {
+                if (!val.file) {
+                    val.name = name;
+                    val.file = file;
+                    val.delete = del;
+                    val.filePath = filePath;
+                }
+                // test for and handle updated values
+                else if (val.file == file) {
+                    val.name = name;
+                    val.delete = del;
+                    val.filePath = filePath;
+                    handleDuplicates = true
+                }
+            })
+        }
+
+
+        // setState and Callback functions
+        this.setState({
+            fileList: handleList,
+        },  // reset function variables
+            () => {
+                if (!handleDuplicates) {
+                    this.props.uploadFile(file, handleList)
+                }
+                handleDuplicates = false;
+            })
+    }
+
+
+
+    render() {
+
+        return (
+            <div>
+                <FormGroup>
+                    {this.state.fileList.map(val => {
+                        return <MultiFileComponent className="multi-file-component"
+                            handleUpload={(file, name, del, filePath) => this.handleUpload(file, name, del, filePath)}
+                            errorText={this.props.errorText}
+                            addError={this.state.addError}
+                            del={(file) => this.del(file)}
```

Comment:
`this.del` isn't defined? But seems like it should be instead of overloading handleUpload to do both